tensorflow/python/ops/array_ops.py

```diff
@@ -1021,6 +1021,14 @@ def unstack(value, num=None, axis=0, name="unstack"):
       num = value_shape[axis].value
   if num is None:
     raise ValueError("Cannot infer num from shape %s" % value_shape)
+  elif isinstance(num, ops.Tensor):
+    if num.shape.ndims == 0:
```

Comment:
The shape may not be known at graph construction time. If the rank is unknown statically, use tf.rank and tf.assert to insert a runtime check in the graph.

Reply:
Please add unit tests that test this with both static and dynamic shapes.

Reply:
@rmlarsen  the shape you said is in 'num'. right??

Reply:
@rmlarsen  I fixed.